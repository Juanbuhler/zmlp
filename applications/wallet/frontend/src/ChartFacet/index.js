import PropTypes from 'prop-types'
import SuspenseBoundary from '../SuspenseBoundary'

import chartShape from '../Chart/shape'

import Card, { VARIANTS as CARD_VARIANTS } from '../Card'

import ChartForm from '../ChartForm'

import ChartFacetContent from './Content'

const ChartFacet = ({ chart, chartIndex, dispatch }) => {
  if (!chart.attribute) {
    return (
      <Card
        variant={CARD_VARIANTS.DARK}
        header=""
        content={
          <ChartForm
            chart={chart}
            chartIndex={chartIndex}
            dispatch={dispatch}
          />
        }
      />
    )
  }

  return (
    <Card
      variant={CARD_VARIANTS.DARK}
      header={chart.attribute}
      content={
        <SuspenseBoundary>
          <ChartFacetContent chart={chart} />
        </SuspenseBoundary>
      }
    />
  )
}

ChartFacet.propTypes = {
  chart: PropTypes.shape(chartShape).isRequired,
  chartIndex: PropTypes.number.isRequired,
  dispatch: PropTypes.func.isRequired,
}

export default ChartFacet
