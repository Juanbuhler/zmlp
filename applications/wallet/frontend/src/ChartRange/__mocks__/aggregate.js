const aggregate = [
  {
    id: '972a8ab5-cdcb-4eea-ada7-f1c88d997fed',
    results: {
      count: null,
      min: 180,
      max: 8525.0,
      avg: 899.5689655172414,
      sum: 521750.0,
    },
  },
]

export default aggregate
