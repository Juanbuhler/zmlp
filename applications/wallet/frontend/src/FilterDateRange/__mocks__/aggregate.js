const aggregate = {
  results: {
    min: '2020-05-04',
    max: '2020-07-04',
  },
}

export default aggregate
