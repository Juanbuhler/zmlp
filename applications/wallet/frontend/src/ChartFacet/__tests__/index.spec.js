import TestRenderer, { act } from 'react-test-renderer'

import aggregate from '../__mocks__/aggregate'

import ChartFacet from '..'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const CHART_ID = '972a8ab5-cdcb-4eea-ada7-f1c88d997fed'

jest.mock('../../ChartForm', () => 'ChartForm')

const noop = () => () => {}

describe('<ChartFacet />', () => {
  it('should render properly', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'location.city',
    }

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    require('swr').__setMockUseSWRResponse({ data: aggregate })

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Add Field Filter' })
        .props.onClick()
    })

    expect(mockRouterPush).toHaveBeenCalledWith(
      {
        pathname: '/[projectId]/visualizer/data-visualization',
        query: {
          projectId: PROJECT_ID,
          query: btoa(
            JSON.stringify([
              {
                type: 'facet',
                attribute: 'location.city',
                values: {},
              },
            ]),
          ),
        },
      },
      `/${PROJECT_ID}/visualizer/data-visualization?query=W3sidHlwZSI6ImZhY2V0IiwiYXR0cmlidXRlIjoibG9jYXRpb24uY2l0eSIsInZhbHVlcyI6e319XQ==`,
    )
  })

  it('should render properly without data', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'location.city',
    }

    require('swr').__setMockUseSWRResponse({})

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should add a filter with a value', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'location.city',
    }

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    require('swr').__setMockUseSWRResponse({ data: aggregate })

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    act(() => {
      component.root.findByProps({ 'aria-label': 'Brooklyn' }).props.onClick()
    })

    expect(mockRouterPush).toHaveBeenCalledWith(
      {
        pathname: '/[projectId]/visualizer/data-visualization',
        query: {
          projectId: PROJECT_ID,
          query: btoa(
            JSON.stringify([
              {
                type: 'facet',
                attribute: 'location.city',
                values: { facets: ['Brooklyn'] },
              },
            ]),
          ),
        },
      },
      `/${PROJECT_ID}/visualizer/data-visualization?query=W3sidHlwZSI6ImZhY2V0IiwiYXR0cmlidXRlIjoibG9jYXRpb24uY2l0eSIsInZhbHVlcyI6eyJmYWNldHMiOlsiQnJvb2tseW4iXX19XQ==`,
    )
  })

  it('should add a value to a filter', () => {
    const query = btoa(
      JSON.stringify([
        {
          type: 'facet',
          attribute: 'location.city',
          values: { facets: ['Brooklyn'] },
        },
      ]),
    )

    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID, query },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'location.city',
    }

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    require('swr').__setMockUseSWRResponse({ data: aggregate })

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    act(() => {
      component.root.findByProps({ 'aria-label': 'Zermatt' }).props.onClick()
    })

    expect(mockRouterPush).toHaveBeenCalledWith(
      {
        pathname: '/[projectId]/visualizer/data-visualization',
        query: {
          projectId: PROJECT_ID,
          query: btoa(
            JSON.stringify([
              {
                type: 'facet',
                attribute: 'location.city',
                values: { facets: ['Brooklyn', 'Zermatt'] },
              },
            ]),
          ),
        },
      },
      `/${PROJECT_ID}/visualizer/data-visualization?query=W3sidHlwZSI6ImZhY2V0IiwiYXR0cmlidXRlIjoibG9jYXRpb24uY2l0eSIsInZhbHVlcyI6eyJmYWNldHMiOlsiQnJvb2tseW4iLCJaZXJtYXR0Il19fV0=`,
    )
  })

  it('should not add a duplicate value', () => {
    const query = btoa(
      JSON.stringify([
        {
          type: 'facet',
          attribute: 'location.city',
          values: { facets: ['Brooklyn'] },
        },
      ]),
    )

    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID, query },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'location.city',
    }

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    require('swr').__setMockUseSWRResponse({ data: aggregate })

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    act(() => {
      component.root.findByProps({ 'aria-label': 'Brooklyn' }).props.onClick()
    })

    expect(mockRouterPush).not.toHaveBeenCalled()
  })

  it('should render without an attribute', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
    }

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={noop} />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should delete', () => {
    const mockDispatch = jest.fn()

    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer/data-visualization',
      query: { projectId: PROJECT_ID },
    })

    const chart = {
      id: CHART_ID,
      type: 'facet',
      attribute: 'system.type',
    }

    const component = TestRenderer.create(
      <ChartFacet chart={chart} chartIndex={0} dispatch={mockDispatch} />,
    )

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Delete Chart' })
        .props.onClick()
    })

    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'DELETE',
      payload: {
        chartIndex: 0,
      },
    })
  })
})
