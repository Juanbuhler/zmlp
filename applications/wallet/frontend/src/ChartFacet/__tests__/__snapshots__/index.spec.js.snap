// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartFacet /> should edit 1`] = `
<ChartForm
  chart={
    Object {
      "attribute": "system.type",
      "id": "972a8ab5-cdcb-4eea-ada7-f1c88d997fed",
      "type": "facet",
    }
  }
  chartIndex={0}
  dispatch={[Function]}
  isEditing={true}
  setIsEditing={[Function]}
/>
`;

exports[`<ChartFacet /> should render properly 1`] = `
.emotion-85 {
  height: 100%;
  overflow: hidden;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-84 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #252525;
  border-radius: 2px;
  padding: 16px;
  padding-top: 8px;
}

.emotion-5 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-bottom: 24px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: -8px;
}

.emotion-4:hover button svg {
  opacity: 1;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-right: 8px;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  padding: 6px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1 svg {
  opacity: 0;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-2 {
  width: 2px;
}

.emotion-83 {
  height: 0%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-83 .ErrorBoundary > div {
  background-color: transparent;
  box-shadow: none;
}

.emotion-83 .Loading {
  background-color: transparent;
  box-shadow: none;
}

.emotion-82 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 8px;
  padding-right: 28px;
  font-family: Roboto Condensed;
  color: #b3b3b3;
}

.emotion-77 {
  margin: 0;
  padding: 0;
  list-style: none;
  overflow-y: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-16 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #ffffff;
}

.emotion-16:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-16:hover,
.emotion-16.focus-visible:focus {
  background-color: #0074f538;
  color: #ffffff;
}

.emotion-16:hover svg,
.emotion-16.focus-visible:focus svg {
  color: #ffffff;
}

.emotion-15 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.emotion-9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7 {
  width: 100%;
  background-color: #0273f5;
}

.emotion-8 {
  height: 4px;
  width: 0%;
  border-top: 1px solid #3b3b3b;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 8px;
  font-family: Roboto Mono;
  font-size: 14px;
  line-height: 18px;
}

.emotion-10 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-11 {
  padding-left: 8px;
}

.emotion-14 {
  margin-top: 4px;
  padding: 4px;
  color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-17 {
  width: 67%;
  background-color: #a03dc7;
}

.emotion-18 {
  height: 4px;
  width: 33%;
  border-top: 1px solid #3b3b3b;
}

.emotion-27 {
  width: 34%;
  background-color: #d6680b;
}

.emotion-28 {
  height: 4px;
  width: 66%;
  border-top: 1px solid #3b3b3b;
}

.emotion-37 {
  width: 34%;
  background-color: #ebb52e;
}

.emotion-47 {
  width: 34%;
  background-color: #00bdc1;
}

.emotion-57 {
  width: 34%;
  background-color: #c54a4a;
}

.emotion-67 {
  width: 0%;
  background-color: #ff7867;
}

.emotion-68 {
  height: 4px;
  width: 100%;
  border-top: 1px solid #3b3b3b;
}

.emotion-81 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-80 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Roboto Condensed;
}

.emotion-80:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-80[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-80:hover,
.emotion-80.focus-visible:focus {
  color: #ffffff;
}

.emotion-80:hover svg,
.emotion-80.focus-visible:focus svg {
  opacity: 1;
}

.emotion-80[aria-disabled=true] {
  color: #808080;
}

.emotion-79 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-78 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 4px;
}

<div
  className="emotion-85"
>
  <div
    className="emotion-84"
  >
    <div
      className="emotion-5"
    >
      <div
        className="emotion-4"
      >
        <h3
          className="emotion-0"
        >
          location.city
        </h3>
        <button
          aria-label="Edit Chart"
          className="emotion-1"
          onClick={[Function]}
          title="Edit"
          type="button"
        >
          <svg
            data-file-name="pen.svg"
            height={20}
          />
        </button>
        <div
          className="emotion-2"
        />
        <button
          aria-label="Delete Chart"
          className="emotion-1"
          onClick={[Function]}
          title="Delete"
          type="button"
        >
          <svg
            data-file-name="cross.svg"
            height={20}
          />
        </button>
      </div>
    </div>
    <div
      className="emotion-83"
    >
      <div
        className="emotion-82"
      >
        <div
          className="emotion-6"
        >
          <div>
            LABEL
          </div>
          <div>
            COUNT
          </div>
        </div>
        <ul
          className="emotion-77"
        >
          <li>
            <button
              aria-label="Tyngsboro"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-7"
                    />
                    <div
                      className="emotion-8"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      Tyngsboro
                    </div>
                    <div
                      className="emotion-11"
                    >
                      3
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="Brooklyn"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-17"
                    />
                    <div
                      className="emotion-18"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      Brooklyn
                    </div>
                    <div
                      className="emotion-11"
                    >
                      2
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="Cát Bà"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-27"
                    />
                    <div
                      className="emotion-28"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      Cát Bà
                    </div>
                    <div
                      className="emotion-11"
                    >
                      1
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="La Habana Vieja"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-37"
                    />
                    <div
                      className="emotion-28"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      La Habana Vieja
                    </div>
                    <div
                      className="emotion-11"
                    >
                      1
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="North Glendale"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-47"
                    />
                    <div
                      className="emotion-28"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      North Glendale
                    </div>
                    <div
                      className="emotion-11"
                    >
                      1
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="Saalfelden am Steinernen Meer"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-57"
                    />
                    <div
                      className="emotion-28"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      Saalfelden am Steinernen Meer
                    </div>
                    <div
                      className="emotion-11"
                    >
                      1
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
          <li>
            <button
              aria-label="Zermatt"
              className="emotion-16"
              onClick={[Function]}
              type="button"
            >
              <div
                className="emotion-15"
              >
                <div
                  className="emotion-13"
                >
                  <div
                    className="emotion-9"
                  >
                    <div
                      className="emotion-67"
                    />
                    <div
                      className="emotion-68"
                    />
                  </div>
                  <div
                    className="emotion-12"
                  >
                    <div
                      className="emotion-10"
                    >
                      Zermatt
                    </div>
                    <div
                      className="emotion-11"
                    >
                      0
                    </div>
                  </div>
                </div>
                <div
                  className="emotion-14"
                >
                  <svg
                    data-file-name="filter.svg"
                    height={20}
                  />
                </div>
              </div>
            </button>
          </li>
        </ul>
        <div
          className="emotion-81"
        >
          <button
            aria-label="Add Field Filter"
            className="emotion-80"
            onClick={[Function]}
            type="button"
          >
            <div
              className="emotion-79"
            >
              <div
                className="emotion-78"
              >
                <svg
                  data-file-name="filter.svg"
                  height={20}
                />
              </div>
              Add Field Filter
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChartFacet /> should render properly without data 1`] = `
.emotion-15 {
  height: 100%;
  overflow: hidden;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-14 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #252525;
  border-radius: 2px;
  padding: 16px;
  padding-top: 8px;
}

.emotion-5 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-bottom: 24px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: -8px;
}

.emotion-4:hover button svg {
  opacity: 1;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-right: 8px;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  padding: 6px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1 svg {
  opacity: 0;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-2 {
  width: 2px;
}

.emotion-13 {
  height: 0%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-13 .ErrorBoundary > div {
  background-color: transparent;
  box-shadow: none;
}

.emotion-13 .Loading {
  background-color: transparent;
  box-shadow: none;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 8px;
  padding-right: 28px;
  font-family: Roboto Condensed;
  color: #b3b3b3;
}

.emotion-7 {
  margin: 0;
  padding: 0;
  list-style: none;
  overflow-y: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Roboto Condensed;
}

.emotion-10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-10:hover,
.emotion-10.focus-visible:focus {
  color: #ffffff;
}

.emotion-10:hover svg,
.emotion-10.focus-visible:focus svg {
  opacity: 1;
}

.emotion-10[aria-disabled=true] {
  color: #808080;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 4px;
}

<div
  className="emotion-15"
>
  <div
    className="emotion-14"
  >
    <div
      className="emotion-5"
    >
      <div
        className="emotion-4"
      >
        <h3
          className="emotion-0"
        >
          location.city
        </h3>
        <button
          aria-label="Edit Chart"
          className="emotion-1"
          onClick={[Function]}
          title="Edit"
          type="button"
        >
          <svg
            data-file-name="pen.svg"
            height={20}
          />
        </button>
        <div
          className="emotion-2"
        />
        <button
          aria-label="Delete Chart"
          className="emotion-1"
          onClick={[Function]}
          title="Delete"
          type="button"
        >
          <svg
            data-file-name="cross.svg"
            height={20}
          />
        </button>
      </div>
    </div>
    <div
      className="emotion-13"
    >
      <div
        className="emotion-12"
      >
        <div
          className="emotion-6"
        >
          <div>
            LABEL
          </div>
          <div>
            COUNT
          </div>
        </div>
        <ul
          className="emotion-7"
        />
        <div
          className="emotion-11"
        >
          <button
            aria-label="Add Field Filter"
            className="emotion-10"
            onClick={[Function]}
            type="button"
          >
            <div
              className="emotion-9"
            >
              <div
                className="emotion-8"
              >
                <svg
                  data-file-name="filter.svg"
                  height={20}
                />
              </div>
              Add Field Filter
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChartFacet /> should render without an attribute 1`] = `
<ChartForm
  chart={
    Object {
      "id": "972a8ab5-cdcb-4eea-ada7-f1c88d997fed",
      "type": "facet",
    }
  }
  chartIndex={0}
  dispatch={[Function]}
  isEditing={false}
  setIsEditing={[Function]}
/>
`;
